src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -0,0 +1,399 @@
+import datetime
+import json
+import logging
+from typing import List, Dict, Any, Set, Union, Tuple, cast, Optional
+
+from ceph.deployment import translate
+from ceph.deployment.drive_group import DriveGroupSpec
+from ceph.deployment.drive_selection import DriveSelection
+
+import orchestrator
+from orchestrator import OrchestratorError
+from mgr_module import MonCommandFailed
+
+from cephadm.services.cephadmservice import CephadmService
+
+
+logger = logging.getLogger(__name__)
+
+
+class OSDService(CephadmService):
+    def create_from_spec(self, drive_group: DriveGroupSpec) -> str:
```

Comment:
I got a tight loop after deploying some OSDs.

Originally the function was decorated with ` @trivial_completion`, which makes its return value is always not False.
Without the decorator, the return value of the function will be `''` (empty string) for later serve() loops, which make the following statement always return True:

https://github.com/ceph/ceph/blob/6d0b1776eae6b48d8b61d501f75430fbf0677260/src/pybind/mgr/cephadm/module.py#L2163-L2164

And cause a tight loop in `serve()`:

https://github.com/ceph/ceph/blob/6d0b1776eae6b48d8b61d501f75430fbf0677260/src/pybind/mgr/cephadm/module.py#L854-L855

Reply:
The goal of this PR is to keep the code as it is, just moving it around. I've reverted that change. Thanks!